JS/Vardges/FitnesApplication/src/MainPages/FitnessAppContainers/FoodPage/FoodForm.jsx

```diff
@@ -0,0 +1,24 @@
+import React,{useState} from 'react'
+
+function FoodForm({onAdd}) {
+    const [text, setText] = useState("");
+    const [value, setValue] = useState()
+    return (
+        <div>
+            <form  style={{marginBlock: '30px'}} onSubmit={(ev) => {
+                ev.preventDefault();
+                setValue(text);
+                onAdd(text);
+                setText("")
+            }}>
+                <input type="text" value={text}  maxLength={20} onChange={(ev) => {
+                    if(ev.target.value !== ' ')
```

Comment:
Please explain what the following code does?
Please dont tell checks the IF, I am interested on logic its quite incomplete from my point of view

Reply:
this prevents the user from being able to start the word with spaces